### DISCUSSION TITLE
Auto create empty structure based on function argument string?

### DISCUSSION THREAD

**Discussion by netadr** [Vector35/binaryninja-api#2968] [API] [answered]
I have an ELF binary and an associated file that defines debug symbols in the following format:
```
0000000000414141 T ExampleFunction(ExampleStruct*, int)
```
I've been working on a way to import these symbols into Binja using the `debuginfo` APIs, and I've ran into a roadblock while trying to account for the function arguments. Is there any way using the API to have Binja auto-create empty structures (e.g. auto-create an empty `ExampleStruct` structure from `ExampleStruct*, int`)?

**Comment by psifertex** [comment, marked as answer]
The "Working With Types" guide has a section on creating anonymous structures via the API:

https://docs.binary.ninja/guide/type.html#create-anonymous-structuresclassunions

Does that help?
